### ISSUE TITLE
SQLite database creation bug when running cloned repo on Intel MacBook 

### ISSUE THREAD

**Issue by SemanticMikeG** [umbraco/Umbraco.Cloud.Issues#634]
### Issue description

After cloning a development Umbraco 10 site from our Umbraco Cloud account, to my 2018 Intel MacBook Pro running OSX 13.2.1, I get an error when first running the project.  Build is successful but I always get the following error with the SQLite database creation:

```
System. AggregateException has been thrown
"One or more errors occurred. (SQLite Error 1: 'no such table: umbracoDeployDependency'.)"
```

![Screenshot 2023-02-14 at 14 44 07](https://user-images.githubusercontent.com/125378904/218771175-c822c581-14cb-4b46-ba5d-b4fdaff998ee.png)

It appears that the Umbraco.sqlite.db file is missing several tables.

My colleagues can follow the same process on Windows machines and everything works as expected.

If I manually copy over a Umbraco.sqlite.db file from my colleague's machine everything runs as expected for me.

So, for some reason, the SQLite database creation tasks are incomplete when my Mac runs it.

Also, and weirdly, I get the same error if I run everything in a Windows VM (Parallels) on my Mac.

Any advice would be gratefully received! :)

**Comment by sajumb** [comment]
Hi @SemanticMikeG ,
I am sorry about the long wait for an answer. I have asked our support department and they have not had any recent support cases with the same difficulties in cloning a Umbraco 10 site to Umbraco Cloud. But they were referring to this article though https://creativewebspecialist.co.uk/2021/09/07/how-to-run-netcore-umbraco-cms-on-a-macbook/ 

In case you still have an issue you could have a look at this walkthrough.

**Comment by SemanticMikeG** [comment]
Several months on and I'm still getting this issue every time I clone an Umbraco Cloud repo to my MacBook.  After the build error shows, If I examine the SQLite database, it is missing the three 'umbracoDeploy*' tables.  If I paste them in from a copy of a working database (from a Windows build) the build then runs successfully.
The projects I am cloning are all based on a parent Cloud project - not sure if this has any relevance to the issue?

**Comment by ronaldbarendse** [comment]
Hi @SemanticMikeG, do you have a row in the `umbracoKeyValue` table with key `Umbraco.Core.Upgrader.State+Deploy.Migration`? If not, the Deploy migration that runs when the `UmbracoApplicationStartingNotification` is published and Umbraco's runtime level is `Run` (successfully finished install or upgrade) hasn't created these tables yet.

I suspect custom code in a notification handler that runs before this migration is creating/updating schema (like document and data types) or content, which then triggers other notifications that are handled by Deploy to update signatures/dependencies (which then fails because of the missing database tables). Can you check whether this is the case and whether (temporarily) removing the notification handler registrations allows the initial install to complete without the exception?

If you don't have any custom notification handlers, can you please provide the complete exception message (including stack trace), CMS and Deploy versions and what settings are set in the [`Umbraco:CMS:Global`](https://docs.umbraco.com/umbraco-cms/reference/configuration/globalsettings) (specifically whether [`ShowMaintenancePageWhenInUpgradeState` is disabled](https://docs.umbraco.com/umbraco-cms/tutorials/custom-error-page#maintenance-page)) and [`Umbraco:CMS:Unattended`](https://docs.umbraco.com/umbraco-cms/reference/configuration/unattendedsettings) sections?

**Comment by SemanticMikeG** [comment]
Thanks for the above - I forwarded it to our main developer to investigate.

In the meantime, I've done more testing and have found the following:

If I clone the Umbraco Cloud repo onto my Mac-formatted hard-drive I get the 'no such table' error, no matter if I run the project setup in Visual Studio Mac or in Visual Studio Windows (via Parallels).

However, if I clone the repo onto the virtual 'drive:c' in Parallels (which is Windows formatted) I can run the setup successfully in Visual Studio Windows.

So it appears that there is an issue with the Mac-formatted drive format and the repo structure, something that causes the set-up to fail no matter which platform runs it.

Any further suggestions would be gratefully received :)

**Comment by SemanticMikeG** [comment]
Been doing some more testing.  I can get a fresh clone to run using only Mac software if I:

1. Clone the repo onto a Windows EXFAT formatted USB stick.
2. Run the project from the USB stick (which is very slow!) until the Umbraco welcome page appears in my browser.
3. Close the project and copy it onto my normal Mac hard drive.
4. Run it from there (which is fast again).

So it feels like a some initialisation file is getting hidden/ignored by the Mac file system, preventing the database setup from running on first-run.

**Comment by SemanticMikeG** [comment]
FURTHER UPDATE:  Okay, so we've now established that it's only when the project is inside the Mac OSX 'users' folder that SQLite errors occur.  Must be something about the permissions on that folder that cause it to fail.  SQLite3 is pre-installed on OSX so I'm not sure if there's anything I can do to fix the issue.  For now, I have to make sure the first-run happens on an external drive - then I can copy it back into my 'users' folder to continue my usual process.

If anyone has any advice about how to update the permissions and prevent this issue - it might help others. Thanks :)

**Comment by EuanGeleziunas** [comment]
Hi @SemanticMikeG 
Bit late to the party, but did you manage to solve this?

I'm running into the issue. Created a project on my windows desktop all works fine. Then I've cloned the project onto my mac and can't get it working for the life in me. The project on my mac is currently inside the users folder.

**Comment by EuanGeleziunas** [comment]
@SemanticMikeG Thanks for the speedy reply!
How stupid! I don't have an external drive so back to Windows for now. I did see your last comment but thought I'd ask in case you figured out a way to get it working without having to use an external drive. 

Thanks again! :)

**Comment by SemanticMikeG** [comment]
Just a follow up on this whole issue - I have now discovered that it might actually be related to the disk format.  Recently I started using a new external SSD which was formatted with the relatively new 'APFS'.  I fully expected cloned project to run fine on this, as it is not subject to the 'permissions' of my user folder, but I got the same 'missing tables' message as before.  I know it worked on an older SSD, but that was formatted using 'Mac OS Extended'.  So I formatted that old disk with APFS and the errors started again.  So, I think SQLite has an issue with APFS. I'm now reformatting my new SSD to use the older format.
